packages/flame/lib/src/components/component.dart

```diff
@@ -200,6 +204,10 @@ class Component with Loadable {
     _parent = null;
     nextParent?.add(this);
     nextParent = null;
+
+    if (parentGame is FlameGame) {
+      parentGame.cleanComponent(this);
```

Comment:
I don't understand why we need this, can't the cleanup be done on `BlocComponent#onRemove`?

Reply:
That was one thing I thought too, but BlocComponent is a mixin, so if the user override onRemove themselves, we can't guarantee that the clean up will happen because of how multiple inheritance works.

Also with this, now we have the counterpart for the prepareComponent, which I believe makes sense for FlameGame api

Reply:
Does it make sense to have it on the FlameGame class? All the cleanup is just internal on the component itself right?

Reply:
> Does it make sense to have it on the FlameGame class? All the cleanup is just internal on the component itself right?

It is, but how would we solve the the inheritance problem if we put this on the `onRemove` of the BlocComponent?

Reply:
This method could just live on component and be called `cleanUp` or something, instead of overriding `onRemove`? But I thought that this scenario worked in dart, that if all overridden methods call super both the mixin and the original class method will be called?

Reply:
It will be called only if the mixin is the only mixin that overrides that method, or if it is the last one on the class declaration. So because of that, we can guarantee that it will run.

Reply:
that is an overarching issue with our mixins though, right? for any method, onLoad, update, render, we could have that problem. I think it's just something we have to live with or deal with it, this seems like a hacky workaround

Reply:
If we consider this a Jacky workaround, than 'prepareComponent' is too.

I don't see this as Hacky, it will be worse to have a subscription leaky on games

Reply:
Alright, since we tested and realized that the multiple inheritance is not behaving the same anymore, I removed and added the unsubscription to the component onRemove.